xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -247,6 +247,29 @@
   #end
 #end
 ###############################
+##   DISPLAY TEXT FIELD
+###############################
+#macro (displayTextField $fieldName $defaultValue)
+  #set ($value = $doc.getValue($fieldName, $obj))
+  #set ($prop = $xclass.get($fieldName))
+  #set ($defaultValueHint = "($escapetool.xml($services.localization.render('userprofile.default')))")
+  #if ($isEdit)
+    #set ($escapedId = $escapetool.xml("XWiki.XWikiUsers_0_${fieldName}"))
+    &lt;input type="text" name="$escapedId" id="$escapedId" size="30"
+      #if ($services.localization.get("XWiki.XWikiUsers_${fieldName}.hint"))
+        #set ($escapedHintId = $escapedSelectId + "_hint")
+        aria-describedby="$escapedHintId"
+      #end
+      #if ($value)
+        value="$value"
```

Comment:
Missing escaping.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2981/commits/72af9633ce65cfc64171030cae704c87c6e61501 :+1: